aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs

```diff
@@ -25,213 +21,174 @@ struct Aggregator {
-        "0x1::events_with_aggregators::test_emit_event_v2",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-    ];
-
-    let mut h = MoveHarness::new();
-    let aptos_framework_account = h.aptos_framework_account();
-    let check_status = |status: ExecutionStatus| {
-        assert_matches!(status, ExecutionStatus::Success);
+macro_rules! increment_counter_emit_event {
+    ($s:expr, $v:expr, $w:expr) => {
+        (
+            $s,
+            format!(
+                "events_with_aggregators::test_increment_counter_and_emit_event_v{}",
+                $v
+            ),
+            $w,
+        )
     };
-    let outputs = run_entry_functions(&mut h, &aptos_framework_account, func_names, check_status);
-
-    let c = assert_some!(h.read_resource::<Counter>(
-        aptos_framework_account.address(),
-        parse_struct_tag("0x1::events_with_aggregators::Counter").unwrap(),
-    ));
-    assert_eq!(c.value.value, 8);
-    check_events(outputs);
 }
 
-fn check_failed_output(failed_output: TransactionOutput) {
-    let status = assert_ok!(failed_output.status().as_kept_status());
-    let aggregator_location =
-        ModuleId::new(AccountAddress::ONE, ident_str!("aggregator_v2").to_owned());
-    if let ExecutionStatus::MoveAbort {
-        location: AbortLocation::Module(id),
-        code: EAGGREGATOR_OVERFLOW,
-        info: Some(_),
-    } = status
-    {
-        assert_eq!(id, aggregator_location);
-    } else {
-        unreachable!("Expected Move abort, got {:?}", status);
-    }
-    assert_eq!(failed_output.events().len(), 1);
+#[test_case(1)]
+#[test_case(2)]
+fn test_events_with_snapshots(version: u64) {
+    let data = vec![
+        increment_counter!(),
+        increment_counter!(),
+        emit_event!(version, Some(2)),
+        increment_counter!(),
+        increment_counter!(),
+        increment_counter_emit_event!(SUCCESS, version, Some(5)),
+        increment_counter!(),
+        emit_event!(version, Some(6)),
+        increment_counter!(),
+        increment_counter!(),
+    ];
+    let h = run(data);
+    assert_counter_value_eq!(h, 8);
 }
 
-#[test]
-fn test_events_with_snapshots_not_emitted_on_abort() {
-    let func_names = vec![
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        "0x1::events_with_aggregators::increment_counter",
-        // Overflows here.
-        "0x1::events_with_aggregators::test_increment_counter_and_emit_event_v1",
```

Comment:
do we still test the overflow?

Reply:
yes, it is part of `increment_counter_emit_event!` but it checks for error code on overflow instead of `SUCCESS`

Reply:
oh it's just passed, literally across on the same line )) great!